src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/ExplicitInitializationCheckTest.java

```diff
@@ -74,4 +74,24 @@ public void testTokensNotNull() {
         Assert.assertNotNull(check.getDefaultTokens());
         Assert.assertNotNull(check.getRequiredTokens());
     }
+
+    @Test
+    public void testonlyObjectReferences() throws Exception {
+        final DefaultConfiguration checkConfig =
+            createCheckConfig(ExplicitInitializationCheck.class);
+        checkConfig.addAttribute("onlyObjectReferences", "true");
+        final String[] expected = {
+            "5:20: " + getCheckMessage(MSG_KEY, "bar", "null"),
+            "14:22: " + getCheckMessage(MSG_KEY, "str1", "null"),
+            "14:35: " + getCheckMessage(MSG_KEY, "str3", "null"),
+            "15:9: " + getCheckMessage(MSG_KEY, "ar1", "null"),
```

Comment:
Arrays are objects: 
http://stackoverflow.com/a/32543765/1015848
http://stackoverflow.com/a/12807748/1015848

@subkrish , please make example in Xdoc for array.